pkg/collector/corechecks/snmp/report/report_utils.go

```diff
@@ -29,15 +22,40 @@ func getColumnValueFromSymbol(values *valuestore.ResultValueStore, symbol checkc
 		return nil, err
 	}
 	for index, value := range columnValues {
-		if symbol.ExtractValuePattern != nil {
-			extractedValue, err := value.ExtractStringValue(symbol.ExtractValuePattern)
-			if err != nil {
-				log.Debugf("error extracting value from `%v` with pattern `%v`: %v", value, symbol.ExtractValuePattern, err)
-				continue
-			}
-			value = extractedValue
+		newValue, err := processValue(symbol, value)
+		if err != nil {
+			continue
 		}
-		newValues[index] = value
+		newValues[index] = newValue
 	}
 	return newValues, nil
 }
+
+func processValue(symbol checkconfig.SymbolConfig, value valuestore.ResultValue) (valuestore.ResultValue, error) {
+	if symbol.ExtractValueCompiled != nil {
+		extractedValue, err := value.ExtractStringValue(symbol.ExtractValueCompiled)
+		if err != nil {
+			log.Debugf("error extracting value from `%v` with pattern `%v`: %v", value, symbol.ExtractValueCompiled, err)
+			return valuestore.ResultValue{}, err
+		}
+		value = extractedValue
+	}
+	if symbol.MatchPatternCompiled != nil {
+		strValue, err := value.ToString()
+		if err != nil {
+			log.Debugf("error converting value to string (value=%v):", value, err)
+			return valuestore.ResultValue{}, err
+		}
+
+		if symbol.MatchPatternCompiled.MatchString(strValue) {
+			replacedVal := checkconfig.RegexReplaceValue(strValue, symbol.MatchPatternCompiled, symbol.MatchValue)
+			if replacedVal == "" {
+				return valuestore.ResultValue{}, fmt.Errorf("pattern `%v` failed to match `%v` with template `%s`", symbol.MatchPattern, strValue, symbol.MatchValue)
```

Comment:
If we're here it means that the pattern matched but that the template is not compatible with the regex pattern (wrong number of groups likely?).
Could we maybe have an error message that reflects that?

Reply:
Indeed, the message could be better. Update the message here: 2f3198bd5